src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -111,15 +111,13 @@ public void UpdatePivot()
 
                 return (request, false);
             }
-            else if (NextSlotRange.TryDequeue(out StorageRange slotRange))
+            else if (TryDequeueAndIncrementCounter(NextSlotRange, out StorageRange slotRange, ref _activeStorageRequests))
```

Comment:
Just an  issue with naming/structuring code, this construction makes it less readable. Can you name this method domain specific rather than general purpose? 

Maybe ideas: 
Move this method as local function here for locality/visibility.
Don't pass params, use fields/properties?

Reply:
Done